Codebase: ES LINTER ISSUE

### Describe the bug

ESLINT is not working correctly in the repository. There are two main errors

`Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.json#overrides[0]': Cannot find module '@typescript-eslint/eslint-plugin'
Require stack:
- /Users/leoncydsilva/Documents/GitHub/primeng/__placeholder__.js
Referenced from: /Users/leoncydsilva/Documents/GitHub/primeng/.eslintrc.json
    at Module._resolveFilename (node:internal/modules/cjs/loader:1129:15)
    at Function.resolve (node:internal/modules/cjs/helpers:116:19)
    at Object.resolve (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2337:46)
    at ConfigArrayFactory._loadPlugin (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3404:33)
    at ConfigArrayFactory._loadExtendedPluginConfig (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3224:29)
    at ConfigArrayFactory._loadExtends (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3145:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3086:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3031:20)
    at _normalizeObjectConfigData.next (<anonymous>)
Uncaught exception received.
Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.json#overrides[0]': Cannot find module '@typescript-eslint/eslint-plugin'
Require stack:
- /Users/leoncydsilva/Documents/GitHub/primeng/__placeholder__.js
Referenced from: /Users/leoncydsilva/Documents/GitHub/primeng/.eslintrc.json
    at Module._resolveFilename (node:internal/modules/cjs/loader:1129:15)
    at Function.resolve (node:internal/modules/cjs/helpers:116:19)
    at Object.resolve (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2337:46)
    at ConfigArrayFactory._loadPlugin (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3404:33)
    at ConfigArrayFactory._loadExtendedPluginConfig (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3224:29)
    at ConfigArrayFactory._loadExtends (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3145:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3086:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3031:20)
    at _normalizeObjectConfigData.next (<anonymous>)
[Info  - 15:51:22] ESLint library loaded from: /Users/leoncydsilva/Documents/GitHub/primeng/node_modules/eslint/lib/api.js`

**which is because package **@typescript-eslint/eslint-plugin": "^6.6.0** is missing in repo**

and 

`Error: Error while loading rule '@angular-eslint/template/eqeqeq': You have used a rule which requires '@angular-eslint/template-parser' to be used as the 'parser' in your ESLint config.
Occurred while linting /Users/leoncydsilva/Documents/GitHub/primeng/src/app/components/animate/animate.ts
    at ensureTemplateParser (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@angular-eslint/utils/dist/eslint-plugin-template/parser-services.js:21:15)
    at create (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@angular-eslint/eslint-plugin-template/dist/rules/eqeqeq.js:39:42)
    at Object.create (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/@angular-eslint/eslint-plugin-template/dist/utils/create-eslint-rule.js:20:24)
    at createRuleListeners (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/eslint/lib/linter/linter.js:870:21)
    at /Users/leoncydsilva/Documents/GitHub/primeng/node_modules/eslint/lib/linter/linter.js:1040:110
    at Array.forEach (<anonymous>)
    at runRules (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/eslint/lib/linter/linter.js:977:34)
    at Linter._verifyWithoutProcessors (/Users/leoncydsilva/Documents/GitHub/primeng/node_modules/eslint/lib/linter/linter.js:1330:31)
    at /Users/leoncydsilva/Documents/GitHub/primeng/node_modules/eslint/lib/linter/linter.js:1812:29
    at Array.map (<anonymous>)`

    **which is becuase we are trying to use rule **@angular-eslint/template/eqeqeq** without the required parser** **@angular-eslint/template-parser**

### Environment

.

### Reproducer

_No response_

### Angular version

16.1.2

### PrimeNG version

16.3.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.0.1

### Browser(s)

_No response_

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -1,6 +1,6 @@
 {
     "root": true,
-    "ignorePatterns": ["projects/**/*"],
+    "ignorePatterns": ["src/app/showcase/**/*"],
     "overrides": [
         {
             "files": ["*.ts"],
@@ -32,10 +32,10 @@
                         "suffixes": [""]
                     }
                 ],
-                "@angular-eslint/template/eqeqeq": [
+                "@angular-eslint/directive-class-suffix": [
                     "error",
                     {
-                      "allowNullOrUndefined": true
+                        "suffixes": [""]
                     }
                 ],
                 "@angular-eslint/no-host-metadata-property": "off",
@@ -58,7 +58,14 @@
         {
             "files": ["*.html"],
             "extends": ["plugin:@angular-eslint/template/recommended"],
-            "rules": {}
+            "rules": {
+                "@angular-eslint/template/eqeqeq": [
+                    "error",
+                    {
+                      "allowNullOrUndefined": true
+                    }
+                ]
+            }
         }
     ]
 }

--- a/package.json
+++ b/package.json
@@ -52,6 +52,7 @@
         "@types/jasmine": "~4.3.1",
         "@types/jest": "^29.5.1",
         "@types/node": "^12.20.55",
+        "@typescript-eslint/eslint-plugin": "^6.6.0",
         "chart.js": "3.3.2",
         "codelyzer": "^0.0.28",
         "del": "^7.0.0",

